app/lib/actions/middleware/propagate_output.rb

```diff
@@ -0,0 +1,14 @@
+module Actions
+  module Middleware
+    class PropagateOutput < Dynflow::Middleware
+      def run(*args)
+        if action.input.keys.include?('subaction_output') && action.input['subaction_output']
+          action.input['subaction_output'].each do |key, value|
+            self.action.output[key] = value
+          end
+        end
+        pass(*args)
```

Comment:
wouldn't pass(*args) need to go at the start of the function, so that the run phase runs before it propagates the output?